flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/PendingCheckpoint.java

```diff
@@ -142,6 +145,7 @@ public PendingCheckpoint(
         this.checkpointId = checkpointId;
         this.checkpointTimestamp = checkpointTimestamp;
         this.notYetAcknowledgedTasks = checkNotNull(verticesToConfirm);
+        this.tasksToCommitTo = checkNotNull(tasksToCommitTo);
```

Comment:
nit: one step further would be to store `CheckpointBrief` in `PendingCheckpoint` instead of collections. But that's probably out of scope of this PR

Reply:
I also more tend to store `CheckpointPlan` to `PendingCheckpoint`. We also do it in the second PR since there would be multiple collections to pass. I then change it in this PR and store `CheckpointPlan` in `PendingCheckpoint`.